src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1899,14 +1899,11 @@ void InterpreterSelectQuery::executeSubqueriesInSetsAndJoins(QueryPlan & query_p
 
     const Settings & settings = context->getSettingsRef();
 
-    auto creating_sets = std::make_unique<CreatingSetsStep>(
-            query_plan.getCurrentDataStream(),
-            std::move(subqueries_for_sets),
-            SizeLimits(settings.max_rows_to_transfer, settings.max_bytes_to_transfer, settings.transfer_overflow_mode),
-            *context);
+    if (subqueries_for_sets.empty())
+        return;
```

Comment:
Could we have a situation with `query_info.input_order_info == true`  and empty subqueries_for_sets? If not it's a bit confucing to check it here. Probably we could make it for entire `executeSubqueriesInSetsAndJoins()` in call places.

Reply:
Well, actually this is checked in both call places too, so `subqueries_for_sets` is never empty here. I will remove this check.

And for me it is actually strange that we sort something here